module-apps/ModuleApps.md

```diff
@@ -1,10 +1,179 @@
+2293 ms INFO  [ApplicationTest] ApplicationTest.cpp:InitHandler:27: initializing!
+```
+
+# Async actions on data notifications on changed data in db
+
+Due to the fact that application can be closed before asynchronous data is recieved one should have initialized:
+`AsyncCallbackReceiver(this)`
+To have notifications on changed db one should connect to notifications from the db:
+```
+bus.channels.push_back(sys::BusChannel::ServiceDBNotifications);
+```
+To propagate changes from the database to UI handling one have to pass these from application to windows, if not via presenter it can be done like this:
+``` c++
+if (msgl->messageType == MessageType::DBServiceNotification) {
+    auto msg = dynamic_cast<db::NotificationMessage *>(msgl);
+    if (msg != nullptr) {
+        userInterfaceDBNotification(msgl,
+            [&]([[maybe_unused]] sys::Message *, [[maybe_unused]] const std::string &) {
+                // your code here
+             });
+        return std::make_shared<sys::ResponseMessage>();
+    }
+}
+```
+
+# Asynchronous database queries
+
+You can perform datababase queries via [query interface](../../module-db/queries/README.md). 
+
+To have safe way to perform asynchronous calls there is `app::AsyncQuery` which:
+* creates a query
+* when it's done results are passed to the callback
+* if i.e. UI element is removed its callback is automatically removed
+
+It's defined here: [AsyncTask.hpp](../apps-common/AsyncTask.hpp)
+
+**NOTE** to automatically pass Async db::query data to appliaction you have to:
+* inherit in appliaction/item from `public app::AsyncCallbackReceiver`
+* pass data to said item in DataReceivedHandler via
+`return handleAsyncResponse(resp);` at the end of `DataReceivedHandler()`
+
+# GUI library
+
+GUI library documentation here: [module-gui](../module-gui/README.md)
+Please mind that it has Doxygen documentation explaining it's use for all base elements and functions.
+
+# System actions
+
+System action is a call from service -> application.
+**NOTE** By design applications are running only when required, whereas services are available at all times after started.
+
+
+With it, we can:
+- Let application manager what application should handle action
+- Let user decide what actions said application can handle
+- Change intristic application state on service/app request
+    i.e. disable popups or request to show different app
+- Change extristic application state on service
+    i.e. show popup in application
+
+## What's a popup
+
+**WARNING** In Module OS popup is a Window which can be shown any time on action request.
+**NOTE** attachPopups should be most probably moved to `manifest` and popup enum just removed
+
+Popups weren't designed as part of system, but rather step by step integrated into it. While there should be no issue with partially overflowing popups in the applications and UI it would mean major sourcecode refactoring and planning.
```

Comment:
```suggestion
Popups weren't designed as part of the system, but rather step by step integrated into it. While there should be no issue with partially overflowing popups in the applications and UI it would mean major source code refactoring and planning.
```